mathics/core/atoms.py

```diff
@@ -448,9 +448,13 @@ def sameQ(self, other) -> bool:
         """Mathics SameQ"""
         if isinstance(other, MachineReal):
             return self.value == other.value
-        elif isinstance(other, PrecisionReal):
-            return self.to_sympy() == other.value
-        return False
+        if isinstance(other, PrecisionReal):
+            other_value = other.value
+        else:
+            return False
+        value = self.value
```

Comment:
I think this logic could be simplified by moving it in the "then" branch above. 

Also, I think the docstring here could be more detailed. 

I also see that none of the Comparison operators appear in the docs.

I will put in a PR based on this to start to address this.